### ISSUE TITLE
Allow tern to use an external tool as an analyzer for container images

### ISSUE THREAD

**Issue by nishakm** [vmware/tern#284]
**Describe the Feature**
Have tern use an external analyzer to analyze the filesystem's contents in a container image

**Use Cases**
- Hook up an external file scanner (like scancode)
- Hook up an external database querying tool
- Hook up a security scanner

**Implementation Changes**
- Allow room in the layer object to hold the output of the tool
- Give a command line argument to run the tool
- Implement formulating the subprocess command for tern to invoke and dump the output to the layer object property.
- Report out the layer object property in all the reports available
Might be dependent on #262

**Comment by nishakm** [comment]
Stretch goal - Release 1.0.0 for November

**Comment by nishakm** [comment]
Steps to accomplish:
1. Create a `files_analyzed` property for each layer with appropriate tests
2. Create an `analyzed_output` property for each layer with appropriate tests
3. Create an `extensions.yml` file which can be empty for now
4. Create a report subcommand option called `-x` or `--extend` which will take in a string
5. Create functionality that will create the command to run the given tool, run it against each layer filesystem, and return the result and error. This will be a new file called `passthrough.py` in `report`.
5. In `report.py` for the `execute_docker_image` and `execute_dockerfile` functions, check to see if the `--extend` flag exists, and if it is, call the main function in `passthrough.py` that does all the work.
6. Special for SPDX documents: files_analyzed would be true for each layer. analyzed_output can be dumped into `PackageComment` for now.

**Comment by nishakm** [comment]
#470 created a plugin for cve-bin-tool. We now need something that demonstrates filling up the properties for the images. Possible candidate is scancode. Roadblocks to this is that scancode is supported in python2.7 but not python3. Another candidate is binaryanalysis-ng, which is easier because python3 is supported. It cannot be installed using pip though.

**Comment by nishakm** [comment]
Scancode's develop branch is now fully ported to python3. I will go ahead and try to build a plugin for it. @pombredanne I owe you this :)

**Comment by nishakm** [comment]
The default, yaml and json report now includes scancode data. Closing.
